cmds/dhclient/dhcp6client/dhcp6client.go

```diff
@@ -0,0 +1,128 @@
+package dhcp6client
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/mdlayher/dhcp6"
+	// "golang.org/x/net/icmp"
+	//"golang.org/x/net/ipv6"
+	// "github.com/d2g/dhcp4"
+)
+
+type Client struct {
+	hardwareAddr  net.HardwareAddr //The HardwareAddr to send in the request.
+	ignoreServers []net.IP         //List of Servers to Ignore requests from.
+	timeout       time.Duration    //Time before we timeout.
+	broadcast     bool             //Set the Bcast flag in BOOTP Flags
+	connection    connection       //The Connection Method to use
+}
+
+/*
+*  * Abstracts the type of underlying socket used
+*   */
+type connection interface {
+	Close() error
+	Write(p *dhcp6.Packet, mac net.HardwareAddr) error
+	// WriteNeighborAd(src, dst net.IP, pb []byte) error
+	ReadFrom() ([]byte, error)
+	// SetReadTimeout(t time.Duration) error
+}
+
+func New(haddr net.HardwareAddr, conn connection, timeout time.Duration) (*Client, error) {
+	c := Client{
+		broadcast: true,
+	}
+
+	c.hardwareAddr = haddr
+	c.connection = conn
+	c.timeout = timeout
+	return &c, nil
+}
+
+func (c *Client) Request(mac *net.HardwareAddr) (bool, *dhcp6.Packet, error) {
+	solicitPacket, err := c.SendSolicitPacket(mac)
```

Comment:
Here's my thought:

What we get in cmds/dhclient/dhclient.go is a packet.
1. If everything goes well, the Request() function will return a reply packet from server, in which contains an assigned ipv6 address that we can bind with the interface (done in cmds/dhclient/dhclient.go). 
2. If we fail to get the reply from server, at least we are able to return a packet, the one we wrap and send out, that we can look into in cmds/dhclient/dhclient.go

I followed the idea by d2g/dhcp4client: https://github.com/d2g/dhcp4client/blob/master/client.go#L335, though I don't have objection against changing this logic.